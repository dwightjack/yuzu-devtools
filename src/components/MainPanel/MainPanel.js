import { html } from 'lit-html';
import { component } from 'haunted';
// import * as styles from './MainPanel.styles';

export default function MainPanel() {
  return html`
    <style>
      :host {
        display: block;
        overflow: auto;
        height: 100%;
      }
    </style>
    <div><slot></slot></div>
  `;
}

customElements.define('yzdt-main-panel', component(MainPanel));
