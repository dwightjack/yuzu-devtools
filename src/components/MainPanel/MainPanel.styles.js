import { css } from 'emotion';

export const root = css`
  overflow: auto;
  height: 100%;
`;
