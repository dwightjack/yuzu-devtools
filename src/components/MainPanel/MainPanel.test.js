import MainPanel from './MainPanel';

const ctx = {};

describe('MainPanel', () => {
  test('it renders', () => {
    expect(MainPanel({ ctx, render: () => 'DEMO' })).toMatchSnapshot();
  });

  test('it passes the context to the render fn', () => {
    const render = jest.fn(() => '');
    MainPanel({ ctx, render });
    expect(render).toHaveBeenCalledWith(ctx);
  });
});
