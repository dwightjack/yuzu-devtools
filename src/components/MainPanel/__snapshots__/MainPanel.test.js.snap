// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPanel matches default snapshot 1`] = `
<div
  class="css-1y4ya79"
  style=""
>
  DEMO
  
</div>
`;
